### PR TITLE
gh-125517: Fix unreachable code warnings in `_testembed.c`

### PR DESCRIPTION
**Pull request by sobolevn**
CC @vstinner as the author of https://github.com/python/cpython/commit/ef9d54703f1f21d2fde2fe3ef49d4cf7c8b38f06

* Issue: gh-125517

### CODE REVIEW

**Review by vstinner — COMMENTED** [review]
Please update also the documentation: https://docs.python.org/dev/c-api/init_config.html#id1

Yeah, it's annoying, it's not possible to declare err_msg in the `error` label :-(

**Comment by sobolevn** [comment]
I like the braces idea! Will do. Thanks for the quick review!

File: Doc/c-api/init_config.rst [hunk 1]
```diff
@@ -1823,14 +1823,16 @@ return ``-1`` on error:
         PyInitConfig_Free(config);
         return 0;
 
-        // Display the error message
-        const char *err_msg;
     error:
```
Comment (vstinner) [inline Doc/c-api/init_config.rst:1826]: You might add a comment to explain this uncommon braces style.

**Review by vstinner — APPROVED** [review]
LGTM

**Comment by sobolevn** [comment]
Thanks! Addressed the last comment as well.

File: Doc/c-api/init_config.rst [hunk 2]
```diff
@@ -1823,14 +1823,18 @@ return ``-1`` on error:
...
-        (void)PyInitConfig_GetError(config, &err_msg);
-        printf("PYTHON INIT ERROR: %s\n", err_msg);
-        PyInitConfig_Free(config);
+        {
+            // Display the error message
+            // This uncommon braces style is used, because you cannot make
+            // goto targets point to variable declarations.
```
Comment (vstinner) [inline Doc/c-api/init_config.rst:1830]: You may put the comment before the brace, rather than here.

**Pull request merged**